src/packages/version-check/src/index.ts

```diff
@@ -0,0 +1,291 @@
+import { TruffleColors } from "@ganache/colors";
+import http2 from "http2";
+import Conf from "conf";
+import { default as semverDiff } from "semver/functions/diff";
+import { default as semverValid } from "semver/functions/valid";
+import { default as semverClean } from "semver/functions/clean";
+import { default as semverGte } from "semver/functions/gte";
+
+export type VersionCheckConfig = {
+  packageName: string;
+  enabled: boolean;
+  url: string;
+  ttl: number;
+  latestVersion: string;
+  latestVersionLogged: string;
+};
+
+export class VersionCheck {
+  protected ConfigManager;
+  protected _config;
+  protected _logger;
+  protected _currentVersion;
+
+  constructor(
+    currentVersion: string,
+    config?: VersionCheckConfig,
+    logger?: Function
+  ) {
+    // Creates a new config, or reads existing from disk
+    this.ConfigManager = new Conf({
+      configName: process.env.TEST ? "testConfig" : "config", // config is the package default
+      defaults: {
+        ...VersionCheck.DEFAULTS,
+        ...config
+      }
+    });
+
+    // pulls the config out of the manager, lays optional props over top
+    this._config = { ...this.ConfigManager.get("config"), ...config };
+
+    // If config was passed in, save changes
+    if (config) this.saveConfig();
```

Comment:
defaults would only affect the value on the file creation. Successively setting defaults will still update the schema property, but it will not change the value saved to disk. 

If you are OK with https://github.com/trufflesuite/ganache/pull/3285/commits/54166b59a226cb05f3a6154d567131150a579569#diff-d3c52bfdae42da5ce88cd5db96d272177e44283f952b3545816b03f08a2c8b85R37 I can remove `defaults` entirely and should only write on first run, when new conf params are passed in or when the value is changed through the `set` method.

Reply:
I made the change to save on startup, even if the defaults are used. 

Previously, it would not unless a value for `config` was passed in. However, it does `getLatestVersion` at the end of the constructor. It will save if a version is returned. 

It will, (should), never report on the first run. Technically, ganache could be installed and not run until a newer version is released. But VC won't know that version number until it's first check in with the server since we won't be shipping it with a `latestVersion` set. That would require bumping with each build.